pages/api/roles/index.ts

```diff
@@ -68,12 +69,32 @@ function handler(req: NextApiRequest, res: NextApiResponse) {
   }
 }
 
-// TODO: Optional Search query. Will be done in another PR to unblock others.
-
 async function handleGet(req: NextApiRequest, res: NextApiResponse<ApiResponse<RoleListData[]>>) {
+  const { companyId, searchQuery } = req.query;
+
+  const queryCompanyId = isInteger(companyId as string) ? Number(companyId) : undefined;
+
+  const searchWords = (searchQuery as string) ? (searchQuery as string).split(' ') : [];
+  const queryYear = searchWords.filter(isInteger).map(Number)[0];
```

Comment:
A bit far fetched but a internship from 5 Guys / Forever 21 could set year with the wrong number (e.g. `Forever 21 Summer Intern 2022`)
```suggestion
  const queryYear = searchWords.filter(isInteger).map(Number).filter(x => x > MIN_YEAR)[0];
```